packages/ogmios/src/ogmiosToCore/tx.ts

```diff
@@ -65,319 +53,305 @@ const mapPoolParameters = (poolParameters: Schema.PoolParameters): Cardano.PoolP
-      collaterals: isAlonzoOrAbove(kind) ? (tx as Schema.TxAlonzo).body.collaterals.map(mapTxIn) : undefined,
-      fee: tx.body.fee,
-      inputs: tx.body.inputs.map(mapTxIn),
-      mint: isMaryOrAbove(kind) ? mapMint(tx as Schema.TxMary) : undefined,
-      outputs: tx.body.outputs.map(mapTxOut),
-      requiredExtraSignatures: isAlonzoOrAbove(kind)
-        ? (tx as Schema.TxAlonzo).body.requiredExtraSignatures.map(Crypto.Ed25519KeyHashHex)
-        : undefined,
-      scriptIntegrityHash: isAlonzoOrAbove(kind) ? mapScriptIntegrityHash(tx as Schema.TxAlonzo) : undefined,
-      validityInterval: isShelleyTx(kind)
-        ? undefined
-        : mapValidityInterval((tx as Schema.TxAlonzo).body.validityInterval),
-      withdrawals: Object.entries(tx.body.withdrawals).map(([key, value]) => ({
-        quantity: value,
-        stakeAddress: Cardano.RewardAccount(key)
-      }))
+      certificates: tx.certificates?.map(mapCertificate),
+      collaterals: tx.collaterals?.map(mapTxIn),
+      fee: tx.fee?.lovelace ?? mapByronTxFee(tx), // You were here. a common tx has fee but byron does not
+      inputs: tx.inputs.map(mapTxIn),
+      mint: mapMint(tx),
+      outputs: tx.outputs.map(mapTxOut),
+      requiredExtraSignatures: tx.requiredExtraSignatories?.map(Crypto.Ed25519KeyHashHex),
+      ...(tx.scriptIntegrityHash && { scriptIntegrityHash: Crypto.Hash32ByteBase16(tx.scriptIntegrityHash) }),
+      ...(tx.validityInterval && { validityInterval: mapValidityInterval(tx.validityInterval) }),
+      ...(tx.withdrawals && {
+        withdrawals: Object.entries(tx.withdrawals).map(([key, { lovelace }]) => ({
+          quantity: lovelace,
+          stakeAddress: Cardano.RewardAccount(key)
+        }))
+      })
     },
     id: Cardano.TransactionId(tx.id),
-    inputSource: isAlonzoOrAbove(kind)
-      ? Cardano.InputSource[(tx as Schema.TxAlonzo).inputSource]
-      : Cardano.InputSource.inputs,
+    inputSource: Cardano.InputSource[tx.spends],
     witness: {
-      bootstrap: tx.witness.bootstrap.map(mapBootstrapWitness),
-      datums: isAlonzoOrAbove(kind) ? mapWitnessDatums(tx as Schema.TxAlonzo) : undefined,
-      redeemers: isAlonzoOrAbove(kind)
-        ? Object.entries((tx as Schema.TxAlonzo).witness.redeemers).map(([key, value]) => mapRedeemer(key, value))
-        : undefined,
-      scripts: [...Object.values(tx.witness.scripts).map(mapScript)],
+      bootstrap: tx.signatories.map(mapBootstrapWitness),
+      ...(tx.datums && { datums: mapWitnessDatums({ datums: tx.datums }) }),
+      ...(tx.redeemers && { redeemers: Object.entries(tx.redeemers).map(([key, value]) => mapRedeemer(key, value)) }),
+      ...(tx.scripts && { scripts: [...Object.values(tx.scripts).map(mapScript)] }),
       signatures: new Map(
-        Object.entries(tx.witness.signatures).map(([key, value]) => [
+        // REVIEW: 'witness.bootstrap' and 'witness.signatures' were both flattened to 'signatories'. How should we handle this?
```

Comment:
REVIEW: clarify flattened 'witness.bootstrap' and 'witness.signatures'

Reply:
Bootstrap witness will have `chainCode` and `attributes`, so we can map this without loss

Reply:
@mirceahasegan apply the opposite filter to bootstrap witnesses